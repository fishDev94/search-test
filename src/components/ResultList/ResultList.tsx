import type { Comment } from "../../types";
import Card from "../UI/Card/Card";

import { paginate, getTotalPages } from "../../utils/pagination";
import { useState } from "react";

import styles from "./result-list.module.scss";
import { mapComponent } from "../../utils/mapComponents";
import Paginator from "../Paginator/Paginator";

export default function ResultList({ data = [] }: { data?: Comment[] }) {
  const [currentPage, setCurrentPage] = useState(1);

  const pages = getTotalPages(data);

  return (
    <div className={styles.result_list}>
      <Paginator
        currentPage={currentPage}
        totalPages={pages}
        setPage={setCurrentPage}
      />
      {mapComponent(paginate(data, currentPage), Card, "comment")}
    </div>
  );
}
