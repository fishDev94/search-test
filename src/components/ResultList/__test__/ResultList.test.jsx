import { render, screen } from "@testing-library/react";
import ResultList from "../ResultList";

describe("ResultList", () => {
  const mockData = [
    {
      id: 1,
      name: "User-test-1",
      body: "Comment 1",
      email: "test-mail@mail.com",
    },
    {
      id: 2,
      name: "User-test-2",
      body: "Comment 2",
      email: "test-mail@mail.com",
    },
  ];

  it("matches snapshot with data", () => {
    const { container } = render(<ResultList data={mockData} />);
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot without data", () => {
    const { container } = render(<ResultList />);
    expect(container).toMatchSnapshot();
  });

  it("renders paginator and cards when data is provided", () => {
    render(<ResultList data={mockData} />);

    const paginatorText = screen.getByText(/Page 1 of/i);
    expect(paginatorText).to.exist;

    const comment1 = screen.getByText("Comment 1");
    const comment2 = screen.getByText("Comment 2");
    expect(comment1).to.exist;
    expect(comment2).to.exist;
  });

  it("renders no cards when data is empty", () => {
    render(<ResultList data={[]} />);

    const comment1 = screen.queryByText("Comment 1");
    const comment2 = screen.queryByText("Comment 2");
    expect(comment1).to.not.exist;
    expect(comment2).to.not.exist;
  });
});
