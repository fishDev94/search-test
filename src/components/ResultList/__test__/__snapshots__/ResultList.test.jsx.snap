// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResultList > matches snapshot with data 1`] = `
<div>
  <div
    class="_result_list_fc25aa"
  >
    <div
      class="_paginator_387b14"
    >
      <button
        class="_app_button_1f8b9b _inactive_1f8b9b"
        disabled=""
        name="app_button"
      >
        Prev
      </button>
      <span>
        Page 
        1
         of 
        1
      </span>
      <button
        class="_app_button_1f8b9b _inactive_1f8b9b"
        disabled=""
        name="app_button"
      >
        Next
      </button>
    </div>
    <div
      class="_card_c3be90 _default_c3be90"
    >
      <h2>
        User-test-1
      </h2>
      <p
        class="_card__email_c3be90"
      >
        test-mail@mail.com
      </p>
      <p>
        Comment 1
         
      </p>
    </div>
    <div
      class="_card_c3be90 _default_c3be90"
    >
      <h2>
        User-test-2
      </h2>
      <p
        class="_card__email_c3be90"
      >
        test-mail@mail.com
      </p>
      <p>
        Comment 2
         
      </p>
    </div>
  </div>
</div>
`;

exports[`ResultList > matches snapshot without data 1`] = `
<div>
  <div
    class="_result_list_fc25aa"
  >
    <div
      class="_paginator_387b14"
    >
      <button
        class="_app_button_1f8b9b _inactive_1f8b9b"
        disabled=""
        name="app_button"
      >
        Prev
      </button>
      <span>
        Page 
        1
         of 
        0
      </span>
      <button
        class="_app_button_1f8b9b "
        name="app_button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
